File: Python/program51.py
def count_factors(i_no):
    i_frequency = 0

    if i_no < 0:
        i_no = -i_no

    # We use integer division (//) because range() requires integers
    for i_cnt in range(1, (i_no // 2) + 1):
        if (i_no % i_cnt) == 0:
            i_frequency += 1

    return i_frequency

if __name__ == "__main__":
    i_value = int(input("Enter the number : \n"))
    i_ret = count_factors(i_value)
    print(f"Number of factors are : {i_ret}")

File: Java/program51.java
//Develop a C program that accepts an integer and returns the total
//count of its factors by iterating up to half of the number's value.
import java.util.Scanner;

public class program51 {

    public static void main (String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iRet = 0;

        System.out.println("Enter an nos ");
        int iValue = sobj.nextInt();
        iRet = Display(iValue);
        System.out.println(iRet);

        sobj.close();
        
    }

    public static int Display(int iNo)
    {
        int count =0;
        for (int i = 1 ; i<=iNo/2 ;i++)
        {
            if(iNo%i == 0)
            {
                count++;
            }
        }
        return count;

    }
    
}
